router/binders.go

```diff
@@ -195,3 +241,54 @@ func responseContentType(responseContentType string, supportedTypes ContentTypes
 	}
 	return defaultContentType, fmt.Errorf("%w: %s", UnsupportedResponseContentTypeErr, responseContentType)
 }
+
+func validateParamsAndPopulateDefaults(ctx *Context, in string) (*openapi3filter.RequestValidationInput, error) {
+	input := requestValidationInput(ctx)
+	parameters := utils.Filter(utils.Map(ctx.Operation.Parameters, func(p *openapi3.ParameterRef) *openapi3.Parameter {
+		return p.Value
+	}), func(p *openapi3.Parameter) bool { return p.In == in })
+
+	for _, param := range parameters {
+		if err := openapi3filter.ValidateParameter(ctx.Request.Context(), input, param); err != nil {
+			return nil, err
+		}
+	}
+
+	// after processing params input is populated with defaults
+	return input, nil
+}
+
+func requestValidationInput(ctx *Context) *openapi3filter.RequestValidationInput {
```

Comment:
Consider: pass the "in String" argument into this method and build "path" of "query" accordingly. Do you need to provide all params when validating only path parameters or only query parameters?

Reply:
Actually, I would like to refactor it in the future so it will run for the entire request (instead of currently for every segment separately) I prefer to keep it this way so I can obtain RequestValidationInput easily from the context.